python/pyspark/sql/functions.py

```diff
@@ -2075,9 +2075,10 @@ class PandasUDFType(object):
 def udf(f=None, returnType=StringType()):
```

Comment:
Do we need to just add a parameter for deterministic? Adding it to the end is OK to PySpark without breaking the existing app? cc @ueshin

Reply:
I followed what was done for scala UDF, where this parameter is not added, but there is a method to add it. If we add a parameter here, I'd then suggest to add it also to the scala API.

Reply:
Scala and Python are different, because that is also for JAVA API.

Reply:
@gatorsmile, however, wouldn't it be better to keep them consistent if possible?

Reply:
I am saying this because I had few talks about this before and I am pretty sure we usually keep them as same whenever possible.

Reply:
Using `asNondeterministic ` is not straightforward to users. In Scala sides, we have no choice for avoiding breaking the API. Anyway, I am fine to keep it as now